src/rest_api/classes/gene/widgets/location.clj

```diff
@@ -0,0 +1,50 @@
+(ns rest-api.classes.gene.widgets.location
+  (:require
+    [rest-api.classes.gene.sequence :as sequence-fns]
+    [rest-api.classes.gene.generic :as generic]
+))
+
+(defn genetic-position [gene]
+  (let [segment (sequence-fns/get-longest-segment gene)
+        gene-map (:gene/map gene)
+        chr (if (nil? gene-map) nil (:map/id (:gene.map/map gene-map)))
```

Comment:
@a8wright In clojure, using keyword on nil returns nil, as desired, so you don't need a special case here.  `(:map/id (:gene.map/map gene-map)` should be sufficient. So are the case with the next few lines.